### PR TITLE
make s3 dependency optional

### PR DESCRIPTION
**Pull request by talIguaz**

### CODE REVIEW

File: storey/utils.py [hunk 1]
```diff
@@ -105,4 +105,19 @@ def _split_path(path):
...
     return fsspec.filesystem(schema), parsed_url.path
+
+
+def load_dependencies(schema):
+    if schema == "s3":
+        try:
+            import s3fs  # noqa: F401
+        except ImportError:
+            raise StoreyMissingDependencyError(
+                "s3 packages are missing, use pip install mlrun[s3]"
```
Comment (gtopper) [inline storey/utils.py]: mlrun?

File: storey/utils.py [hunk 2]
```diff
@@ -105,4 +105,19 @@ def _split_path(path):
 def url_to_file_system(url):
     parsed_url = urlparse(url)
     schema = parsed_url.scheme.lower()
+    load_dependencies(schema)
     return fsspec.filesystem(schema), parsed_url.path
+
+
+def load_dependencies(schema):
```
Comment (gtopper) [inline storey/utils.py]: Generic name, but actually this is only for s3.

Comment (talIguaz) [reply storey/utils.py]: will be used for other fs in the future, like azure

File: storey/utils.py (hunk 1 above)
Comment (talIguaz) [reply storey/utils.py]: 🙈

File: setup.py [hunk 3]
```diff
@@ -23,6 +23,10 @@ def load_deps(file_name):
 
 install_requires = load_deps('requirements.txt')
 tests_require = load_deps('dev-requirements.txt')
+extras_require = {
+    "s3": ["s3fs~=0.5.2"],
```
Comment (gtopper) [inline setup.py]: `~=0.5`?

File: storey/utils.py (hunk 2 above)
Comment (gtopper) [reply storey/utils.py]: Probably would've been better named as `load_filesystem_dependencies`  or `load_fs_dependencies `

**Review by gtopper — APPROVED** [review]

**Pull request merged**